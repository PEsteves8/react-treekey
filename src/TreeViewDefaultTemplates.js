import React from "react";
import styles from "./defaultTreeStyles";

export let defaultToggle = (isExpanded) => {
  const svgStyles = {
    fill: "white",
    strokeWidth: 0,
    transformOrigin: "center",
    transform: `rotate(${isExpanded ? "90deg" : "0"})`,
  };

  let { height, width, points } = styles.nodeToggleSvg;

  return (
    <svg height={height} width={width}>
      <polygon points={points} style={svgStyles} />
    </svg>
  );
};

export let defaultHeader = (node) => {
  return <span style={{ paddingLeft: "7px" }}>{node.name}</span>;
};
