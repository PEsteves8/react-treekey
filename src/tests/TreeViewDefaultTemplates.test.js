import { defaultHeader, defaultToggle } from "../TreeViewDefaultTemplates";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

const mockNode = {
  name: "some node",
};

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("<TreeViewDefaultTemplates>", () => {
  test("should match default snapshot", () => {
    act(() => {
      render(defaultHeader(mockNode), container);
    });

    expect(container.innerHTML).toMatchSnapshot();
  });

  test("should match default snapshot", () => {
    act(() => {
      render(defaultToggle(true), container);
    });

    expect(container.innerHTML).toMatchSnapshot();

    act(() => {
      render(defaultToggle(false), container);
    });

    expect(container.innerHTML).toMatchSnapshot();
  });
});
