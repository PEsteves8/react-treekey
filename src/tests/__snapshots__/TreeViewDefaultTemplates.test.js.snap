// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeViewDefaultTemplates> should match default snapshot 1`] = `"<span style=\\"padding-left: 7px;\\">some node</span>"`;

exports[`<TreeViewDefaultTemplates> should match default snapshot 2`] = `"<svg height=\\"10\\" width=\\"8\\"><polygon points=\\"0,0,0,10,8,5\\" style=\\"fill: white; stroke-width: 0; transform-origin: center; transform: rotate(90deg);\\"></polygon></svg>"`;

exports[`<TreeViewDefaultTemplates> should match default snapshot 3`] = `"<svg height=\\"10\\" width=\\"8\\"><polygon points=\\"0,0,0,10,8,5\\" style=\\"fill: white; stroke-width: 0; transform-origin: center; transform: rotate(0);\\"></polygon></svg>"`;
